docs/guides/tutorials/chatgpt_bot.rst

```diff
@@ -0,0 +1,2305 @@
+extending the vector and specifying the desired length. OpenAI embeddings have
+length of 1,536, so that's what we use in our schema for this custom scalar.
+
+Now, the ``Section`` type:
+
+.. code-block:: sdl
+    :caption: dbschema/default.esdl
+
+    …
+      type Section {
+        required content: str;
+        required tokens: int16;
+        required embedding: OpenAIEmbedding;
+
+        index ext::pgvector::ivfflat_cosine(lists := 3)
+          on (.embedding);
+      }
+    …
+
+The ``Section`` contains properties to store the content, a count of tokens,
+and the embedding, which is of the custom scalar type we created in the
+previous step.
+
+We've also added an index inside the ``Section`` type to speed up queries. In
+order for this to work properly, the index should correspond to the
+``cosine_similarity`` function we're going to use to find sections related to
+the user's question. That corresponding index is ``ivfflat_cosine``.
+
+We are using the value ``3`` for the ``lists`` parameter because best practice
+is to use the number of objects divided by 1,000 for up to 1,000,000 objects.
+Our database will have around 3,000 total objects which falls well under that
+threshold.
+
+In our case indexing does not have much impact, but if you plan to store and
+query a large number of entries, you'll see performance gains by adding this
+index.
+
+Put that all together, and your entire schema file should look like this:
+
+.. code-block:: sdl
+    :caption: dbschema/default.esdl
+
+    using extension pgvector;
+
+    module default {
+      scalar type OpenAIEmbedding extending
+        ext::pgvector::vector<1536>;
+
+      type Section {
+        required content: str;
+        required tokens: int16;
+        required embedding: OpenAIEmbedding;
+
+        index ext::pgvector::ivfflat_cosine(lists := 3)
+          on (.embedding);
+      }
+    }
+
+We apply this schema by creating and running a migration.
+
+.. code-block:: bash
+
+    $ edgedb migration create
+    $ edgedb migrate
+
+.. note::
+
+    In this tutorial we will regenerate all embeddings every time we run the
+    embeddings generation script, wiping all data and saving new ``Section``
+    objects for all of the documentation. This might be a reasonable approach
+    if you don't have much documentation, but if you have a lot of
+    documentation, you may want a more sophisticated approach that operates on
+    only documentation sections which have changed.
+
+    You can achieve this by saving content checksums and section paths as part
+    of your ``Section`` objects. The next time you run generation, compare the
+    section's current checksum with the one you stored in the database, finding
+    it by its path. You don't need to generate embeddings and update the
+    database for a given section unless the two checksums are different
+    indicating something has changed.
+
+    If you decide to go this route, add properties your ``Section`` as shown
```

Comment:
we miss here "to your Section" I guess

In general users can save path or title, there should be a property that is unique, I used paths to allow having the same file titles under different folders... not sure is it easy to explain that it doesn't have to be `only path`

Reply:
Good catch!

I went ahead and generalized this advice too to let readers know they can use any unique identifier.